spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/IntegrationFlowFunctionSupport.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.function;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.reactivestreams.Publisher;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.function.context.FunctionCatalog;
+import org.springframework.cloud.function.context.catalog.FunctionInspector;
+import org.springframework.cloud.function.core.FluxSupplier;
+import org.springframework.cloud.stream.converter.CompositeMessageConverterFactory;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ *
+ * @author Oleg Zhurakousky
+ *
+ * @since 2.1
+ */
+public class IntegrationFlowFunctionSupport {
+
+	private final FunctionCatalog functionCatalog;
+
+	private final FunctionInspector functionInspector;
+
+	private final CompositeMessageConverterFactory messageConverterFactory;
+
+	private final FunctionProperties functionProperties;
+
+	/**
+	 *
+	 * @param functionCatalog
+	 * @param functionInspector
+	 * @param messageConverterFactory
+	 * @param functionProperties
+	 */
+	public IntegrationFlowFunctionSupport(FunctionCatalog functionCatalog, FunctionInspector functionInspector,
+			CompositeMessageConverterFactory messageConverterFactory, FunctionProperties functionProperties) {
+
+		Assert.notNull(functionCatalog, "'functionCatalog' must not be null");
+		Assert.notNull(functionInspector, "'functionInspector' must not be null");
+		Assert.notNull(messageConverterFactory, "'messageConverterFactory' must not be null");
+		this.functionCatalog = functionCatalog;
+		this.functionInspector = functionInspector;
+		this.messageConverterFactory = messageConverterFactory;
+		this.functionProperties = functionProperties;
+	}
+
+	/**
+	 * Creates an instance of the {@link IntegrationFlowBuilder} from a {@link Supplier} bean available in the context.
```

Comment:
Our Spring style is *imperative* on methods: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting. Also no blank lines in the method Javadocs.